lib/collection/src/collection_manager/holders/proxy_segment.rs

```diff
@@ -499,13 +502,67 @@ impl SegmentEntry for ProxySegment {
     fn vector_dim(&self) -> usize {
         self.write_segment.get().read().vector_dim()
     }
+
+    fn take_snapshot(&self, snapshot_dir_path: &Path) -> OperationResult<()> {
+        log::info!(
+            "Taking a snapshot of a proxy segment into {:?}",
+            snapshot_dir_path
+        );
+        // extra care is needed to capture outstanding deleted points
+        let deleted_points_guard = self.deleted_points.read();
+        let wrapped_segment_arc = self.wrapped_segment.get();
+        let wrapped_segment_guard = wrapped_segment_arc.read();
+
+        // stable copy of the deleted points at the time of the snapshot
+        let deleted_points_copy = deleted_points_guard.clone();
+
+        // create unique dir. to hold data copy of wrapped segment
+        let copy_target_dir = snapshot_dir_path.join(format!("segment_copy_{}", Uuid::new_v4()));
+        create_dir_all(&copy_target_dir)?;
+
+        // copy proxy segment current wrapped data
+        let full_copy_path = self.copy_segment_directory(&copy_target_dir)?;
+        // snapshot write_segment
+        self.write_segment
+            .get()
+            .read()
+            .take_snapshot(snapshot_dir_path)?;
+
+        // unlock deleted_points as we have a stable copy
+        drop(wrapped_segment_guard);
+        drop(deleted_points_guard);
+
+        // load copy of wrapped segment in memory
+        let mut in_memory_wrapped_segment = load_segment(&full_copy_path)?;
+        let mut segment_version = in_memory_wrapped_segment.version;
+
+        // remove potentially deleted points from wrapped_segment
+        for deleted_point in deleted_points_copy {
+            segment_version += 1;
```

Comment:
it's too arbitrary. We can't guarantee that we won't end up with version much higher than it was actually created in WAL. And if it happens - segment will decline some update operations as if they were already applied.

But it should be safe if we simply take the version from the `write_segment` - it is guaranteed to be higher than anything in wrapped segment and does not exceed WAL at the same time